### PR TITLE
Fix default dynamic-asssets directory

### LINKED ISSUE #6475: Ensogl loader default options are outdated
The defaults in ensogl config options in loader groups are set to:
```
"wasmUrl": {
          "value": "pkg.wasm",
          "description": "The URL of the WASM pkg file generated by ensogl-pack.",
          "primary": false
        },
        "jsUrl": {
          "value": "pkg.js",
          "description": "The URL of the JS pkg file generated by ensogl-pack.",
          "primary": false
        },
        "assetsUrl": {
          "value": "assets",
          "description": "The URL of the dynamic assets directory.",
          "primary": false
        },
```
This is no longer true as `assetsUrl` should have `dynamic-assets`. We need to check and fix values of  `wasmUrl` (either `pkg.wasm` or `pkg-opt.wasm`).

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

Set the default value correctly, and use it. Fixes #6475.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] ~The documentation has been updated, if necessary.~
- [ ] ~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] ~Unit tests have been written where possible.~
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by somebody1234 — APPROVED** [review]
it's two lines, i don't think there's *that* much that could possibly go wrong

should probably still get *some* QA though. i think cloud dashboard can be exempt from the QA, since it always explicitly specifies URLs

File: app/ide-desktop/lib/content/src/index.ts [hunk 1]
```diff
@@ -143,7 +143,6 @@ class Main implements AppRunner {
                 loader: {
                     wasmUrl: 'pkg-opt.wasm',
                     jsUrl: 'pkg.js',
-                    assetsUrl: 'dynamic-assets',
```
Comment (wdanilo) [inline app/ide-desktop/lib/content/src/index.ts]: `jsUrl` is not needed either because it is the same as default.

**Review by wdanilo — APPROVED** [review]
This is super small change but Q/A from cloud team is needed. CC @PabloBuchu @indiv0 @somebody1234 . Especially, now we have partial object and we need to get if deep option merging behaves as it should.

**Comment by somebody1234** [comment]
@wdanilo Options are merged using `Object.assign`, so it's a shallow merge, not a deep merge.
However, I'm not sure it's worth spending the time to make deep merging work as:
- I believe `wasmUrl` should be made consistent everywhere as well, so the merging shouldn't be needed at all, right?
- The cloud frontend provides values for all three URLs anyway

However, if you do think a deep merge is more future-proof, we can do that manually instead: `{ ...inputConfig, loader: { current options, ...inputConfig.loader } }`

**Comment by wdanilo** [comment]
@somebody1234 I believe options are merged deeply. I implemented it in App:

image

The only shallow merge is the merge in the content/src/index, but I believe we can live with it for now, or ... do what you proposed above - I like it, let's do it this way!

**Comment by somebody1234** [comment]
QA:
:heavy_check_mark: `./run gui watch`
:heavy_check_mark: `./run gui watch`, new dashboard (all options for gui passed through URL params)
:heavy_check_mark: `./run gui watch`, alternate startup entry
:heavy_check_mark: `npm run watch-dashboard`. note: opening projects currently does not work there (not a bug). i can investigate later whether it's possible now that the new API to start projects has been merged in
:heavy_check_mark: `enso.AppImage`, passing invalid `-loader.*-url` results in one failed fetch (indicating that only one option is being changed, so deep merge is working correctly)
:heavy_check_mark: `enso.AppImage`, invalid `js-url` + invalid `wasm-url` results in two failed fetches
:heavy_check_mark: `enso.AppImage`, opening local projects
:heavy_check_mark: `./run ide watch`, opening local projects
  - ends at "successfully ending" without opening a window
  - pre-built cli has the same issue. i guess it might be the one built by `./run`, which would mean both commands are running the same binary
  - not sure this is related however, as this PR does not touch the build CLI. it might also be just an issue with cache as usual?
  - ~~actually `watch.ts` might be exiting early~~ upon further investigation it seems that `watch.ts` is not being run at all
  - seems to be broken on every commit :( i'll just do a clean
  - edit of the edit: still broken no matter how i clean. i'll just reboot
  - fixed by reboot

note: opening cloud projects not tested as they are slow to open for me, and they are passing the same set of options

**Comment by wdanilo** [comment]
GUI Lint is failing on CI. @somebody1234 do you know why it could happen?

**Comment by somebody1234** [comment]
@wdanilo looks like this is the error:
```
java.io.FileNotFoundException: built-distribution/enso-engine-2023.2.1-dev-linux-amd64/enso-2023.2.1-dev/lib/Standard/AWS/2023.2.1-dev/package.yaml (No such file or directory)
```

it does look familiar to me so i think it's unlikely that any GUI changes would be the cause of this. i think the same issue has been occurring for a lot of other PRs

upon investigation it seems like [the AWS library](https://github.com/enso-org/enso/tree/402c95b181d58443562f730b7d857ceb0b49bb1b/distribution/lib/Standard/AWS) is new. so something must not be happening, leading to the AWS library not getting built.

from a cursory glance it doesn't seem like anything was missed for the AWS library in [build.sbt](https://github.com/enso-org/enso/blob/402c95b181d58443562f730b7d857ceb0b49bb1b/build.sbt) but it might be good for someone more familiar with libs to double check. worst case i think we can just clean runners again and hope for the best

**Comment by wdanilo** [comment]
@somebody1234 I'm talking about GUI/CI/Lint:

image

**Comment by somebody1234** [comment]
ah sorry, my bad.
@wdanilo that should be fixed by that one PR to fix CI? [specifically, this line](https://github.com/enso-org/enso/blob/0a8f80959f65c9891ff8cffb68c5d9a7e6cd527c/app/ide-desktop/lib/types/modules.d.ts#L5)

so i think this *should* be fixed by merging develop into this branch?

**Comment by wdanilo** [comment]
@somebody1234 I was initially thinking the same, but this PR is marked as `CI: ready to merge`, which means it's automatically kept up to date with develop. Github also shows it's up to date. So I'm afraid it's not the case here :(

**Comment by somebody1234** [comment]
@wdanilo actually, iirc `ready to merge` was changed to not keep up to date to avoid overloading CI runners. i've added the `keep up to date` tag. let's see if that fixes lint CI

**Comment by wdanilo** [comment]
> @wdanilo actually, iirc `ready to merge` was changed to not keep up to date to avoid overloading CI runners. i've added the `keep up to date` tag. let's see if that fixes lint CI

Oh, did not catch this change! Thanks and if this is the case, I apologize for the confusion here!

**Comment by somebody1234** [comment]
[seems like lint is still failing](https://github.com/enso-org/enso/actions/runs/4889821834/jobs/8728677511?pr=6537), but the cause is `prettier` having issues with `CHANGELOG.md` (?!). it seems like the CI fails before `typecheck` and `eslint` have a chance to run though, so there *could* still be issues with those tasks

**Comment by somebody1234** [comment]
seems like that lint was failing [on the PR it was introduced in](https://github.com/enso-org/enso/actions/runs/4886194544/jobs/8721220666). might need a hotfix to fix that?

**Comment by somebody1234** [comment]
lint success :D

**Comment by wdanilo** [comment]
Perfect, thanks so much @somebody1234 for taking care of it!

**Pull request merged**